content-review/impl/urkund/src/main/java/org/sakaiproject/contentreview/urkund/UrkundAccountConnection.java

```diff
@@ -0,0 +1,129 @@
+ * tools or provisioned user spaces (such as different campuses, etc).
+ *
+ * A large portion of this was factored out of UrkundReviewService where it
+ * originally occurred.
+ *
+ *
+ */
+public class UrkundAccountConnection {
+	private static final Logger log = LoggerFactory.getLogger(UrkundAccountConnection.class);
+	
+	private final static String DEFAULT_API_URL = "https://secure.urkund.com/api/";
+	private final static int DEFAULT_TIMEOUT = 180000;
+
+	private String apiURL = "";
+	private String receiverAddress = "";
+	private String username = null;
+	private String password = null;
+	private int connTimeout = 0; // Default to 0, no timeout.
+
+	public void init() {
+
+		log.info("init()");
+
+		receiverAddress = serverConfigurationService.getString("urkund.address");
+		username = serverConfigurationService.getString("urkund.username");
+		password = serverConfigurationService.getString("urkund.password");
+
+		apiURL = serverConfigurationService.getString("urkund.apiURL", DEFAULT_API_URL);
+
+		// Timeout period in ms for network connections (default 180s). Set to 0 to disable timeout.
+		connTimeout = serverConfigurationService.getInt("urkund.networkTimeout", DEFAULT_TIMEOUT);
+
+	}
+
+	/*
+	 * Utility Methods below
+	 */	
+	public UrkundSubmissionData uploadFile(String submitterEmail, String externalId, String filename, byte[] filecontent, String mimeType) {
+		String jsonResponse = UrkundAPIUtil.postDocument(apiURL, submitterEmail, externalId, filename, filecontent, mimeType, receiverAddress, username, password, connTimeout);
+		return getSubmissionData(externalId, jsonResponse);
+	}
+	public List<UrkundSubmissionData> getReports(String externalId) {
+		String jsonResponse = UrkundAPIUtil.getFileInfo(apiURL, externalId, receiverAddress, username, password, connTimeout);
+		return getSubmissionDataList(jsonResponse);
+	}
+	
+	private UrkundSubmissionData getSubmissionData(String externalId, String jsonResponse){
+		ObjectMapper mapper = new ObjectMapper();
+		try {
+			JsonNode root = mapper.readTree(jsonResponse);
+			//this should not happen
+			if(root.isArray() && root.size() > 0) {
+				root = root.get(0);
+			}
+			UrkundSubmissionData ret = mapper.treeToValue(root, UrkundSubmissionData.class);
+			if(ret.getExternalId() != null && ret.getExternalId().equals(externalId)){
+				return ret;
+			}
+		} catch (JsonProcessingException e) {
+			log.error("Error getting submission data : JsonProcessingException", e);
+		} catch (IOException e) {
+			log.error("Error getting submission data : IOException", e);
+		} catch (Exception e) {
+			log.error("Error getting submission data : Exception", e);
+		}
+		return null;
+	}
+	
+	private List<UrkundSubmissionData> getSubmissionDataList(String jsonResponse){
+		ObjectMapper mapper = new ObjectMapper();
+		try {
+			JsonNode root = mapper.readTree(jsonResponse);
+			UrkundSubmissionData[] ret = mapper.treeToValue(root, UrkundSubmissionData[].class);
+			return Arrays.asList(ret);
```

Comment:
Arrays.asList() returns an Arrays.ArrayList(array) which is backed by the original array and has certain limitations and should be treated as immutable (most of time). This doesn't appear to match the requirement.